controller/openebs/reconciler.go

```diff
@@ -42,34 +46,23 @@ func (h *reconcileErrHandler) handle(err error) {
 		"Failed to reconcile OpenEBS %s %s: %+v",
 		h.openebs.GetNamespace(), h.openebs.GetName(), err,
 	)
-
-	conds, mergeErr :=
-		k8s.MergeStatusConditions(
-			h.openebs, types.MakeOpenEBSReconcileErrCond(err),
-		)
-	if mergeErr != nil {
-		glog.Errorf(
-			"Failed to reconcile OpenEBS %s %s: Can't set status conditions: %+v",
-			h.openebs.GetNamespace(), h.openebs.GetName(), mergeErr,
-		)
-		// Note: Merge error will reset the conditions which will make
-		// things worse since various controllers will be reconciling
-		// based on these conditions.
-		//
-		// Hence it is better to set response status as nil to let metac
-		// preserve old status conditions if any.
-		h.hookResponse.Status = nil
-	} else {
-		// response status will be set against the watch's status by metac
-		h.hookResponse.Status = map[string]interface{}{}
-		h.hookResponse.Status["phase"] = types.OpenEBSStatusPhaseError
-		h.hookResponse.Status["conditions"] = conds
-	}
+	// response status will be set against the watch's status by metac
+	h.hookResponse.Status = map[string]interface{}{}
+	h.hookResponse.Status["phase"] = types.OpenEBSStatusPhaseFailed
+	h.hookResponse.Status["reason"] = err.Error()
 	// this will stop further reconciliation at metac since there was
 	// an error
 	h.hookResponse.SkipReconcile = true
 }
 
+// This method is being used for setting the success response
+// against watch's status by metac.
+func (h *reconcileSuccessHandler) handle() {
+	// response status will be set against the watch's status by metac
+	h.hookResponse.Status = map[string]interface{}{}
+	h.hookResponse.Status["phase"] = types.OpenEBSStatusPhaseOnline
```

Comment:
Please verify if string(types.OpenEBSStatusPhaseOnline) is better to avoid continuous reconciliation i.e. hot loop path.